aws/utility.sh

```diff
@@ -540,6 +554,25 @@ function addJSONAncestorObjects() {
   runJQ "${pattern}" < "${file}"
 }
 
+# -- KMS --
+function decrypt_kms_string() { 
+  local value="$1"; shift
+
+  local tmpdir="$( getTempDir "kms_XXX" )"
```

Comment:
Suggest using the new pushTempDir function - it tries to keep all the temp dirs under a single root dir

Reply:
Couldn't find pushTempDir...?

Reply:
Ah - yes. Added it in the stuff I'm working on. Dno't worry about it for now. But basically rather than a heap of tmpdirs in the top level, I've added a pushTempDir command like getTempDir but it basically maintains a stack of tmpdirs. If its used, the routine then PopTempDir's when its done. Just means we only get one top level dir in tmp for a given build rather than a million of them